nibabel/tests/test_funcs.py

```diff
@@ -150,3 +157,22 @@ def test_closest_canonical():
     assert_true(img is xyz_img)
     # it's still not diagnonal
     assert_raises(OrientationError, as_closest_canonical, img, True)
+
+    # an axis swap
+    aff = np.diag([1, 0, 0, 1])
+    aff[1, 2] = 1; aff[2, 1] = 1
+    img = Nifti1Image(arr, aff)
+    img.header.set_dim_info(0, 1, 2)
+
+    xyz_img = as_closest_canonical(img)
+    assert_false(img is xyz_img)
+    # Check both the original and new objects
+    assert_true(img.header.get_dim_info() == (0, 1, 2))
+    assert_true(xyz_img.header.get_dim_info() == (0, 2, 1))
+    out_arr = xyz_img.get_data()
+    assert_array_equal(out_arr, np.transpose(arr, (0, 2, 1, 3)))
+
+    # same axis swap but with None dim info (except for slice dim)
+    img.header.set_dim_info(None, None, 2)
```

Comment:
This probably the same question as before, but why isn't the correct answer `(None, 2, None)` after this permutation?